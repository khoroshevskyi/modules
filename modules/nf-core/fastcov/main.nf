process FASTCOV {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/fe/fe768c866b4ae4f2c8948ea7a274ce66d590eedb1cf967495dbd0fb84643a7e2/data': 'community.wave.seqera.io/library/fastcov:0.1.3--84def91a6ef27f61' }"

    input:
    tuple val(meta), path(bam), path(index)
    val(file_ext)

    output:
    tuple val(meta), path("${prefix}.${file_ext}"), emit: coverage_plot
    path "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    file_ext = file_ext.isEmpty() ? 'png' : file_ext[0]
    args = args.isEmpty() ? '' : args[0]
    def VERSION = '0.1.3' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    fastcov.py \\
      -o ${prefix}.${file_ext} \\
      ${args} \\
      ${bam}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastcov: $VERSION
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.1.3' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    touch ${prefix}.${file_ext}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastcov: $VERSION
    END_VERSIONS
    """
}
