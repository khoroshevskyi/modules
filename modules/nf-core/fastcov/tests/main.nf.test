nextflow_process {

    name "Test Process FASTCOV"
    script "../main.nf"
    process "FASTCOV"

    tag "modules"
    tag "modules_nfcore"
    tag "fastcov"

    test("sarscov2 - bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true)],
                    [ file(params.modules_testdata_base_path + '/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)],
                    [ 'png' ],
                    [ '' ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions,
                                file(process.out.coverage_plot.get(0).get(1)).name
                                ).match()
                    }
            )
        }

    }

    test("sarscov2 - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    [],
                    [],
                    [],
                    []
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
