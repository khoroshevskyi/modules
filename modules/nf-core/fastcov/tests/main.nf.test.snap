{
    "sarscov2 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.[]:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
                ],
                "coverage_plot": [
                    [
                        {
                            "id": "test"
                        },
                        "test.[]:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-07T11:20:28.928329396"
    },
    "sarscov2 - bam": {
        "content": [
            [
                "versions.yml:md5,1b47585c2aa870d7ffad63209dfe6d2b"
            ],
            "test.png"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T14:10:58.359505"
    }
}